pennylane_lightning/core/src/simulators/lightning_gpu/gates/tests/Test_StateVectorCudaManaged_NonParam.cpp

```diff
@@ -0,0 +1,1174 @@
+                   "[StateVectorCudaManaged_Nonparam]", float, double) {
+    using PrecisionT = TestType;
+    const std::size_t num_qubits = 3;
+    std::mt19937 re{1337};
+
+    //`values[i]` on the host will be copy the `indices[i]`th element of the
+    // state vector on the device.
+    SECTION("Set state vector with values and their corresponding indices on "
+            "the host") {
+        auto init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits);
+        auto expected_state = init_state;
+
+        for (size_t i = 0; i < Pennylane::Util::exp2(num_qubits - 1); i++) {
+            std::swap(expected_state[i * 2], expected_state[i * 2 + 1]);
+        }
+
+        StateVectorCudaManaged<TestType> sv{num_qubits};
+        sv.CopyHostDataToGpu(init_state.data(), init_state.size());
+
+        using index_type =
+            typename std::conditional<std::is_same<PrecisionT, float>::value,
+                                      int32_t, int64_t>::type;
+        // The setStates will shuffle the state vector values on the device with
+        // the following indices and values setting on host. For example, the
+        // values[i] is used to set the indices[i] th element of state vector on
+        // the device. For example, values[2] (init_state[5]) will be copied to
+        // indices[2]th or (4th) element of the state vector.
+        std::vector<index_type> indices = {0, 2, 4, 6, 1, 3, 5, 7};
+
+        std::vector<std::complex<PrecisionT>> values = {
+            init_state[1], init_state[3], init_state[5], init_state[7],
+            init_state[0], init_state[2], init_state[4], init_state[6]};
+
+        sv.template setStateVector<index_type>(values.size(), values.data(),
+                                               indices.data(), false);
+
+        CHECK(expected_state == Pennylane::Util::approx(sv.getDataVector()));
+    }
+}
+
+TEMPLATE_TEST_CASE("StateVectorCudaManaged::SetStateVectorwith_thread_setting",
+                   "[StateVectorCudaManaged_Nonparam]", float, double) {
+    using PrecisionT = TestType;
+    const std::size_t num_qubits = 3;
+    std::mt19937 re{1337};
+
+    SECTION("SetStates with a non-default GPU thread setting") {
+        auto init_state =
+            createRandomStateVectorData<PrecisionT>(re, num_qubits);
+        auto expected_state = init_state;
+
+        for (size_t i = 0; i < Pennylane::Util::exp2(num_qubits - 1); i++) {
+            std::swap(expected_state[i * 2], expected_state[i * 2 + 1]);
+        }
+
+        StateVectorCudaManaged<TestType> sv{num_qubits};
+        sv.CopyHostDataToGpu(init_state.data(), init_state.size());
+
+        using index_type =
+            typename std::conditional<std::is_same<PrecisionT, float>::value,
+                                      int32_t, int64_t>::type;
+
+        std::vector<index_type> indices = {0, 2, 4, 6, 1, 3, 5, 7};
+
+        std::vector<std::complex<PrecisionT>> values = {
```

Comment:
I assume this line is getting replaced or inlined into the contents of the SV. Maybe add an exclusion here for coverage

Reply:
Done!